src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/Egress/AzureBlob/AzureBlobEgressProviderTests.cs

```diff
@@ -314,6 +407,45 @@ private string GetFileSHA256(string filePath)
             return BitConverter.ToString(sha.ComputeHash(fileStream));
         }
 
+        private void ValidateQueue(List<BlobItem> blobs, List<QueueMessage> messages, int expectedCount)
+        {
+            Assert.Equal(expectedCount, messages.Count);
+            Assert.Equal(expectedCount, blobs.Count);
+
+            HashSet<string> blobNames = new(blobs.Select((b) => b.Name));
+            foreach (QueueMessage message in messages)
+            {
+                Assert.Contains(DecodeQueueMessageBody(message.Body), blobNames);
+            }
+        }
+
+        private string ConstructBlobContainerSasToken(BlobContainerClient containerClient)
+        {
+            // Requires:
+            // - Add for UploadAction.ProvideUploadStream
+            // - Write for UploadAction.WriteToProviderStream
+            BlobSasBuilder sasBuilder = new(
+                BlobContainerSasPermissions.Add | BlobContainerSasPermissions.Write,
+                DateTimeOffset.MaxValue)
+            {
+                StartsOn = DateTime.UtcNow
```

Comment:
When working with live Azure storage accounts, they recommend that you backstart the start time by some buffer amount, usually 5 minutes if I recall correctly, to avoid time synchronization discrepancies. I know this running on the same machine so hopefully that isn't an issue, but maybe adding the buffer here and for queue SAS token generation could avoid this possibility.